eksctl/libsonnet/nodegroup.jsonnet

```
@@ -1,34 +1,54 @@
-// This file is referenced by ../template.jsonnet. It declares an object
-// representing a node group that is used as a common foundation for all
-// our node groups.
-//
+/*
+  This file is imported by ../template.jsonnet and its generated files.
 
-// Make Auto Scaling Group (ASG) tags for given set of k8s labels
-local makeCloudLabels(labels) = {
-  ['k8s.io/cluster-autoscaler/node-template/label/%s' % key]: labels[key]
-  for key in std.objectFields(labels)
-};
+  Exports an object representing a node group's default fields, and is supposed
+  to be merged with other objects.
 
-# Make ASG tags for given set of k8s taints
-local makeCloudTaints(taints) = {
-  ['k8s.io/cluster-autoscaler/node-template/taint/%s' % key]: taints[key]
-  for key in std.objectFields(taints)
-};
+  Note that the "name" output field is declared to be calculated based on hidden
+  fields that objects merging into this can influence.
+*/
 
+// Exported object
 {
-  name: '',
+  // If using spot instances where a Auto Scaling Group (ASG) has multiple
+  // instances associated with it, label using the first instance type.
+  // FIXME: Clarify the limitations of picking one of multiple instance types
+  local instanceType = if 'instanceType' in $ then $.instanceType else $.instancesDistribution.instanceTypes[0],
+  local escapedInstanceType = std.strReplace(instanceType, ".", "-"),
```

Let's move this comment that was in cluster's jsonnet before, here since we've changed where we do the name escaping.
```suggestion
  // // NodeGroup names can't have a '.' in them, while instanceTypes always have a .
  local escapedInstanceType = std.strReplace(instanceType, ".", "-"),
```

---

Thank you for reviewing @GeorgianaElena, resolved by f1b7c8a